docs/design/include/mcmini/spy/intercept/wrappers.hpp

```diff
@@ -0,0 +1,47 @@
+#ifndef WRAPPERS_H
+#define WRAPPERS_H
+
+#include <pthread.h>
+
+
+// Adjust the transition type enum based on your requirements
+enum TransitionType { MutexInit, /* Add other transition types as needed */ };
+
+struct MCTransition {
+  // Include necessary information about the transition
+  // (This is a placeholder; you may extend it as needed)
+  std::shared_ptr<void> data;
```

Comment:
I've never seen `std::shared_ptr<void>` but it looks like it [works](http://shanekirk.com/2017/11/c-stdshared_ptrvoid/) but not in the way I think we'd be hoping for (`data = void* type`)

There is a double-free issue as it is defined I believe though. When `data` goes out of scope, `delete T` is called. But `data` was created with `mmap`. You could either define a customer deleter that calls `munmap` instead or else just use `void *` and `munmap` somewhere appropriate`.

If this is meant for C though we couldn't do this anyway (see the comment on `MCSharedTransition`).